File: week-02/day-4/Workshop 13.4.2023/src/ShoppingList2.java
import java.util.HashMap;


public class ShoppingList2 {
    public static void main(String[] args) {
        HashMap<String, Double> prices = new HashMap<>();
        prices.put("Milk", 1.07);
        prices.put("Rice", 1.59);
        prices.put("Eggs", 3.14);
        prices.put("Cheese", 12.60);
        prices.put("Chicken Breasts", 9.40);
        prices.put("Apples", 2.31);
        prices.put("Tomato", 2.58);
        prices.put("Potato", 1.75);
        prices.put("Onion", 1.10);

        HashMap<String, Integer> Bob = new HashMap<>();
        Bob.put("Milk", 3);
        Bob.put("Rice", 2);
        Bob.put("Eggs", 2);
        Bob.put("Cheese", 1);
        Bob.put("Chicken Breasts", 4);
        Bob.put("Apples", 1);
        Bob.put("Tomato", 2);
        Bob.put("Potato", 1);

        HashMap<String, Integer> Alice = new HashMap<>();
        Alice.put("Rice", 1);
        Alice.put("Eggs", 5);
        Alice.put("Chicken Breasts", 2);
        Alice.put("Apples", 1);
        Alice.put("Tomato", 10);

        double bob = 0;
        double alice = 0;







        // How much does Bob pay?
        for (String food : Bob.keySet()) {
            if (prices.containsKey(food)) {
                double a = prices.get(food) * Bob.get(food);
                bob = bob + a;
            }
        }
        System.out.println(bob);

        //How much does Alice pay? x
        for (String food : Alice.keySet()) {
            if (prices.containsKey(food)) {
                double a = prices.get(food) * Alice.get(food);
                alice = alice + a;
            }
        }
        System.out.println(alice);

        //Who buys more Rice?
        if (Bob.get("Rice") > Alice.get("Rice")) {
            System.out.println("Bob");
        } else if (Bob.get("Rice") < Alice.get("Rice")) {
            System.out.println("Alice");
        } else System.out.println("no one");

        //Who buys more Potato?
        try {
            if (Bob.get("Potato") > Alice.get("Potato")) {
                System.out.println("Bob");
            } else if (Bob.get("Potato") < Alice.get("Potato")) {
                System.out.println("Alice");
            } else System.out.println("no one");
        } catch (Exception e) {
            if (Bob.containsKey("Potato")) {
                System.out.println("Bob");
            } else if (Alice.containsKey("Potato")) {
                System.out.println("Alice");
            } else System.out.println("no one");
        }

        //Who buys more Ham?
        try {
            if (Bob.get("Ham") > Alice.get("Ham")) {
                System.out.println("Bob");
            } else if (Bob.get("Ham") < Alice.get("Ham")) {
                System.out.println("Alice");
            } else System.out.println("no one");
        } catch (Exception e) {
            System.out.println("no one");
        }

        //Who buys more Apples?x
        if (Bob.get("Apples") > Alice.get("Apples")) {
            System.out.println("Bob");
        } else if (Bob.get("Apples") < Alice.get("Apples")) {
            System.out.println("Alice");
        } else System.out.println("no one");

        //Who buys more of different products?x
        int AliceCounter = 0;
        int BobCounter = 0;

        for (String AKey : Alice.keySet()) {
            AliceCounter++;
        }
        for (String BKey : Bob.keySet()) {
            BobCounter++;
        }

        if (BobCounter > AliceCounter) {
            System.out.println("Bob");
        } else if (BobCounter < AliceCounter) {
            System.out.println("Alice");
        } else System.out.println("no one");


       // Who buys more items?(more pieces)x
        int AliceValue = 0;
        int BobValue = 0;

        for (String AValue : Alice.keySet()) {
            AliceValue = AliceValue + Alice.get(AValue);
        }

        for (String BValue : Bob.keySet()) {
            BobValue = BobValue + Bob.get(BValue);
        }

        if (BobValue > AliceValue) {
            System.out.println("Bob");
        } else if (BobValue < AliceValue) {
            System.out.println("Alice");
        } else System.out.println("no one");
    }
}

File: week-07/day-2/Workshop 16.5.2023/shopping_list2.py
prices = {}
prices['Milk'] = 1.07
prices['Rice'] = 1.59
prices['Eggs'] = 3.14
prices['Cheese'] = 12.60
prices['Chicken Breasts'] = 9.40
prices['Apples'] = 2.31
prices['Tomato'] = 2.58
prices['Potato'] = 1.75
prices['Onion'] = 1.10

Bob = {}
Bob['Milk'] = 3
Bob['Rice'] = 2
Bob['Eggs'] = 2
Bob['Cheese'] = 1
Bob['Chicken Breasts'] = 4
Bob['Apples'] = 1
Bob['Tomato'] = 2
Bob['Potato'] = 1

Alice = {}
Alice['Rice'] = 1
Alice['Eggs'] = 5
Alice['Chicken Breasts'] = 2
Alice['Apples'] = 1
Alice['Tomato'] = 10

value_Bob = 0
for item in prices:
    if item in Bob:
        value_Bob = value_Bob + (prices.get(item)*Bob.get(item))
print(value_Bob)

value_Alice = 0
for item in prices:
    if item in Alice:
        value_Alice = value_Alice + (prices.get(item)*Alice.get(item))
print(value_Alice)

if Alice.get('Rice') > Bob.get('Rice'):
    print('Alice')
else:
    print('Bob')

try:
    if Alice.get('Potato') > Bob.get('Potato'):
        print('Alice')
    else:
        print('Bob')
except:
    print('Bob')

try:
    if Alice.get('Ham') > Bob.get('Ham'):
        print('Alice')
    else:
        print('Bob')
except:
    print('no one')

try:
    if Alice.get('Apple') > Bob.get('Apple'):
        print('Alice')
    else:
        print('Bob')
except:
    print('no one')

if len(Alice) > len(Bob):
    print('Alice')
else:
    print('Bob')

count_Bob = 0
for item in Bob:
    count_Bob = count_Bob+(Bob.get(item))

count_Alice = 0
for item in Alice:
    count_Alice = count_Alice + (Alice.get(item))

if count_Alice > count_Bob:
    print('Alice')
else:
    print('Bob')